pallets/lbp/src/tests.rs

```diff
@@ -2372,6 +2372,59 @@ fn buy_should_work() {
 	});
 }
 
+#[test]
+fn buy_should_work_when_limit_is_set_above_account_balance() {
+	predefined_test_ext().execute_with(|| {
+		let buyer = BOB;
+		let asset_in = KUSD;
+		let asset_out = BSX;
+
+		//start sale
+		set_block_number::<Test>(11);
+
+		assert_ok!(LBPPallet::buy(
+			Origin::signed(buyer),
+			asset_out,
+			asset_in,
+			10_000_000_u128,
+			u128::MAX,
+		));
+
+		// swap assets
+		set_block_number::<Test>(11);
+		assert_ok!(LBPPallet::buy(
+			Origin::signed(buyer),
+			asset_in,
+			asset_out,
+			10_000_000_u128,
+			u128::MAX,
+		));
+
+		expect_events(vec![
+			Event::BuyExecuted {
+				who: buyer,
+				asset_out: BSX,
+				asset_in: KUSD,
+				amount: 17_894_737,
+				buy_price: 10_000_000,
+				fee_asset: KUSD,
+				fee_amount: 35_860,
+			}
+			.into(),
+			Event::BuyExecuted {
+				who: buyer,
+				asset_out: KUSD,
+				asset_in: BSX,
+				amount: 5_560_301,
+				buy_price: 10_000_000,
+				fee_asset: KUSD,
+				fee_amount: 20_000,
+			}
+			.into(),
+		]);
+	});
+}
+
```

Comment:
I think we should also add negative test cases triggering the `InsufficientAssetBalance` error, both when we buy accumulated and distributed assets.

Reply:
I have just added some two missing cases. Please check the latest commit.